src/main/java/org/aksw/simba/lemming/metrics/single/MaxVertexDegreeMetric.java

```diff
@@ -28,5 +34,186 @@ public double apply(ColouredGraph graph) {
             return graph.getGraph().getMaxOutEdgeDegrees();
         }
     }
+    
+    /** Stores the previously computed values in UpdateMetricResult object and checks if degrees is not updated for vertices in candidate set.
+     * @param graph - input graph.
+     * @param graphOperation - boolean value indicating graph operation. ("true" for adding an edge and "false" for removing an edge)
+     * @param triple - Edge on which graph operation is performed.
+     * @param newMetricResult - UpdatableMetricResult object containing the results that should be updated.
+     * @return UpdatableMetricResult object with updated candidate set and metric values.
+     */
+    @Override
+    public UpdatableMetricResult applyUpdatable(ColouredGraph graph, boolean graphOperation, TripleBaseSingleID triple, UpdatableMetricResult newMetricResult) {
+    	MaxVertexDegreeMetricResult metricResultTempObj = (MaxVertexDegreeMetricResult) newMetricResult;
```

Comment:
Reusing the given object like that is not a good idea. It would simply override the results of previous runs and, hence, may change the metric results of other graphs as well.

Reply:
The applyUpdatable metric, in this class, will be called for MaxVertexDegree metric. Hence, the previous result object will be of type MaxVertexDegreeMetricResult. Therefore, I have defined it in this manner. So that it can call methods defined specifically for max vertex degree metric result.

Reply:
You misunderstand my argument. I do not say that you may end up with a result object that has the wrong type. My argument is, that you reuse an existing result and override it. This will lead to problems because you delete the results of other graphs and override them with your results!